From 398c1872178c095b9215474e660497a0eba5b5ff Mon Sep 17 00:00:00 2001
From: Jake <jake@fulu.org>
Date: Tue, 14 Jul 2026 00:28:34 +0100
Subject: [PATCH] More validation of date and year values in Special:Drilldown

Backport of aa6412f5 and 1cb5bbf9 from master (T422774, T428274, CVE-2026-14363, CVE-2026-58521).
---
 drilldown/CargoFilterValue.php | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drilldown/CargoFilterValue.php b/drilldown/CargoFilterValue.php
index 25ddc2b..fabd8b8 100644
--- a/drilldown/CargoFilterValue.php
+++ b/drilldown/CargoFilterValue.php
@@ -46,25 +46,25 @@ class CargoFilterValue {
 			if ( strpos( $fv->text, ' - ' ) > 0 ) {
 				// If there's a dash, assume it's a year range
 				$years = explode( ' - ', $fv->text );
-				$fv->year = $years[0];
-				$fv->end_year = $years[1];
+				$fv->year = (int)$years[0];
+				$fv->end_year = (int)$years[1];
 				$fv->time_period = 'year range';
 			} else {
 				$date_parts = explode( ' ', $fv->text );
 				if ( count( $date_parts ) == 3 ) {
 					[ $month_str, $day_str, $year ] = explode( ' ', $fv->text );
 					$fv->month = CargoDrilldownUtils::stringToMonth( $month_str, $lang );
-					$fv->day = str_replace( ',', '', $day_str );
-					$fv->year = $year;
+					$fv->day = (int)str_replace( ',', '', $day_str );
+					$fv->year = (int)$year;
 					$fv->time_period = 'day';
 				} elseif ( count( $date_parts ) == 2 ) {
 					[ $month_str, $year ] = explode( ' ', $fv->text );
 					$fv->month = CargoDrilldownUtils::stringToMonth( $month_str, $lang );
-					$fv->year = $year;
+					$fv->year = (int)$year;
 					$fv->time_period = 'month';
 				} else {
 					$fv->month = null;
-					$fv->year = $fv->text;
+					$fv->year = (int)$fv->text;
 					$fv->time_period = 'year';
 				}
 			}
-- 
2.43.0

